joomla/joomla-cms
# [5.0] Enforce Strict parsing for both article and category alias …

Pull Request for Issue #32490 .

### Summary of Changes
The router used in 3.9 and used in 4.0 allows for Search Engine URL forgeries (see #32490) for the impact of this (business) vulnerability.
This PR enforces the ID and Alias in the URL to match. If no match, a 404 will be produced (same as in the Modern Router)

This PR is a redo of #32500 (it now also checks the category alias), after several attempted PR's by @PhilETaylor and discussions on these PRs (#32880 #33270 #32879 #32490 and probably missing some) .

In the discussions the preference was shown to fix this in the router, which this PR does.

### Testing Instructions
In article option tab integration switch Remove IDs from URLs to 'No'
_Testing instructions are for com_content, but should be repeated with com_contact and com_newsfeeds_

[A] Test:
1. open on the front-end [yourdomain]/index.php/blog/5-your-modules
2. open on the front-end [yourdomain]/index.php/blog/5-dont-do-any-business-with-me-because-i-suck

Install this PR
[B]Repeat test above

### Actual result BEFORE applying this Pull Request
[A]: 1 > Your Modules page in displayed
[A]: 2 > Your Modules page is displayed (note: the non-existing / fake URL will be indexed by Google because it resolves okay and as such will show in your sites search results)

### Expected result AFTER applying this Pull Request
[B]: 1> Your Modules page is displayed
[B]: 2 > a 404 is thrown which is the correct behavior as this page doesn't exist and should NEVER resolve okay!


### Documentation Changes Required

cc: @PhilETaylor @Hackwar
